pynblint/repository.py

```diff
@@ -67,6 +70,50 @@ def large_file_paths(self) -> List[Path]:
                     large_files.append(file_path)
         return large_files
 
+    def _get_dependencies(self) -> set:
+        dependencies = set()
+        for root, dirs, files in os.walk(self.path):
+            for f in files:
+                # check if exist a requiremets.txt file
+                if f.endswith("requirements.txt"):
+                    try:
+                        with open((Path(root) / Path(f)), "r") as fi:
+                            file_row = fi.read()
+                            # print(file_row)
+                            tmp = file_row.split("\n")
+                            for item in tmp:
+                                dependencies.add(item)
+                            # dependencies.add(file_row)
+                            # dependencies.add(file_row)
+                    except Exception as e:
+                        print(e)
+                elif f.endswith("poetry.lock"):
+                    continue
+                elif f.endswith("pyproject.toml"):
+                    continue
+                elif f.endswith("environment.yml"):
+                    continue
+                elif f.endswith("setup.py"):
+                    continue
+                elif f.endswith("Pipfile"):
+                    continue
+        # print(dependencies)
+
+        return dependencies
+
+    def _get_core_dependecies(self) -> set:
+        coredependecies = set()
+        # path = None
+        for i in sys.path:
+            if i.endswith("\\lib\\site-packages"):
+                for root, dirs, files in os.walk(i):
+                    for f in files:
+                        if f.endswith(".py"):
+                            coredependecies.add(f)
```

Comment:
1. as you can read in this [Stack Overflow post](https://stackoverflow.com/a/31384640/4178082), `site-packages` does not contain just Python's core packages, but it's rather "the target directory of manually built Python packages" in general.
2. Is a dynamic check like this really necessary? This seems like a fixed information to me (at least if we consider a single Python version at a time)